test/test_constraints.py

```diff
@@ -69,3 +70,106 @@ def test_unknown_labels_not_in_chunks(num_chunks, chunk_size):
                               random_state=SEED)
 
   assert np.all(chunks[labels < 0] < 0)
+
+
+def test_generate_knntriplets():
+  """Toy example validation of knn triplets construction"""
+  k = 1
+  X = np.array([[0, 0], [2, 2], [4, 4], [8, 8], [16, 16], [32, 32], [64, 64],
+                [128, 128], [256, 256]])
+  y = np.array([1, 1, 1, 2, 2, 2, 3, 3, 3])
+
+  T_test = np.array([[0, 1, 3], [1, 0, 3], [2, 1, 3], [3, 4, 2], [4, 3, 2],
+                     [5, 4, 2], [6, 7, 5], [7, 6, 5], [8, 7, 5]])
+  T = Constraints(y).generate_knntriplets(X, k, k)
+
+  assert np.array_equal(sorted(T.tolist()), sorted(T_test.tolist()))
```

Comment:
Not sure what is the added value of this compared to what you have now in `test_generate_knntriplets_k`?

Reply:
yes, I was thinking the same! I will just leave the other test.